### DISCUSSION TITLE
Prisma, NestJS and Swagger

### DISCUSSION THREAD

**Discussion by VanTanev** [notiz-dev/nestjs-prisma#8] [Q&A]
Is there a way to extract swagger API docs from Prisma types?

**Comment by marcjulian** [comment]
@VanTanev can you give an example?

**Reply by VanTanev** [reply]
So, the best I could do was this:

Given the following schema
```
model User {
  id    Int     @default(autoincrement()) @id
  email String  @unique
  name  String?
  role: Role
}

enum Role {
  Admin
  Moderator
  User
}
```

I did this:
```ts
// User.dto.ts
import { User as IUser, Role } from '@prisma/client'

export class User implements IUser {
  id: number
  email: string
  name: string | undefined
  @ApiProperty({ enum: Role })
  role: Role
}
```

This seems to generate OK with swagger, but I was hoping to avoid having to re-declare the `User` - and now I have 2 different user types floating around, my custom one and the Prisma one.

Edit: Actually, one is a TS type, and the other is an ES6 Class, although it compiles down to a fairly simplistic:
```ts
class User {
  static _OPENAPI_METADATA_FACTORY() {
    return { id: { required: true, type: () => Number }, ... }
  }
}
__decorate([
    (0, swagger_1.ApiProperty)({ enum: client_1.Role}),
    __metadata("design:type", String)
], User.prototype, "role", void 0);
```

And the class can be used mostly as a type, without ever instantiating it at runtime.
